tests/users/test_resend_email.py

```diff
@@ -0,0 +1,151 @@
+import logging
+import uuid
+from django.urls import reverse
+from django.utils import timezone
+from rest_framework.test import APITestCase
+from unittest.mock import patch
+from users.models import User, UserRole
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+class ResendEmailTests(APITestCase):
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.user_role, _ = UserRole.objects.get_or_create(role=UserRole.Role.USER)
+
+    def setUp(self):
+        self.user = User.objects.create(
+            email=f"user_{uuid.uuid4().hex[:10]}@example.com",
+            first_name='Test',
+            last_name='User',
+            role=self.user_role,
+            is_active=False,
+            email_verification_token='oldtoken',
+            email_verification_sent_at=timezone.now()
+        )
+        logger.info("Created test user: %s", self.user.email)
```

Comment:
Remove or reduce logging unless debugging. Test output should generally remain clean.